### PR TITLE
Custom atmos

### PR DESCRIPTION
**Pull request by AdamPlenty**

### CODE REVIEW

**Comment by Loobinex** [comment]
I don't understand why you do it like this. Now you hard-configure where the custom atmos sounds go. Wouldn't it make much more sense to instead of a boolean make it 3 numbers: 'background sound', 'range start' and 'range end'.

This way the feature keeps working even when the sound.dat changes, and it allows the user much more flexibility

**Comment by AdamPlenty** [comment]
> 
> 
> I don't understand why you do it like this. Now you hard-configure where the custom atmos sounds go. Wouldn't it make much more sense to instead of a boolean make it 3 numbers: 'background sound', 'range start' and 'range end'.
> 
> This way the feature keeps working even when the sound.dat changes, and it allows the user much more flexibility

Yes, that probably would make more sense.

**Comment by Loobinex** [comment]
What did you do with that force-push? Keeping feature branches clean is not important, because as soon as they are merged we delete them.

**Comment by AdamPlenty** [comment]
> 
> 
> What did you do with that force-push? Keeping feature branches clean is not important, because as soon as they are merged we delete them.

I tried to get rid of the redundant commits.

**Comment by Loobinex** [comment]
well, now my commit is gone.

**Comment by Loobinex** [comment]
I don't understand why you have the CustomAtmos setting. In the current release there already is a range of atmos sounds, hard coded. Why don't you simply make that configurable, and put the defaults to the current values. 
None of the other config values in the game have a separate setting you need to enable to change them.

Also, don't you want the water drops to be configurable as well?

**Comment by AdamPlenty** [comment]
I'll do that in the next commit. And yes, I'd like the water drops configurable. By 'configurable', you mean enabled/disabled?

**Comment by Loobinex** [comment]
No, I mean why just these water drops. Why not other water drops or custom sounds.
Those other soundfonts, what did they have?

**Comment by AdamPlenty** [comment]
They didn't have a loop of anything fi that's what you mean.

File: src/config.c [hunk 1]
```diff
@@ -750,6 +754,33 @@ short load_configuration(void)
...
+	  case 15: // Atmos
+          for (i=0; i<3; i++)
+          {
+            if (get_conf_parameter_single(buf,&pos,len,word_buf,sizeof(word_buf)) > 0)
+              k = atoi(word_buf);
+            else
+              k = -1;
+            if (k<=0)
+            {
+                CONFWRNLOG("Couldn't recognize setting %d in \"%s\" command of %s file.",
```
Comment (Loobinex) [inline src/config.c]: @AdamPlenty This doesn't seem to work for me. I expect to get this message when I don't have the setting in the log, but I never see it.

Comment (AdamPlenty) [reply src/config.c]: > 
> 
> @AdamPlenty This doesn't seem to work for me. I expect to get this message when I don't have the setting in the log, but I never see it.

I don't know why that would be. Does it work when you have no resolution settings?

Comment (Loobinex) [reply src/config.c]: It does not, no.

**Pull request merged**